ports/raspberrypi/common-hal/audiopwmio/PWMAudioOut.c

```diff
@@ -126,17 +126,14 @@ void common_hal_audiopwmio_pwmaudioout_deinit(audiopwmio_pwmaudioout_obj_t *self
 }
 
 void common_hal_audiopwmio_pwmaudioout_play(audiopwmio_pwmaudioout_obj_t *self, mp_obj_t sample, bool loop) {
-    if (common_hal_audiopwmio_pwmaudioout_get_playing(self)) {
-        common_hal_audiopwmio_pwmaudioout_stop(self);
-    }
```

Comment:
What I think I see is that by removing the "stop", `dma_hw->timer[self->pacing_timer]` is no longer set to 0, causing each play-while-already-playing to allocate a new pacing timer.

If you revise your test program to repeatedly start a new looping sample without closing it first, does it soon fail due to "No DMA pacing timer found"?

I did not actually test anything, I just looked at the code.

Reply:
You are correct; we need to stop if the dma channel is currently playing. So, I've made modifications in common_hal_audiopwmio_pwmaudioout_play() to detect if that is the case and only do the 'stop' if the dma channel is already in use.